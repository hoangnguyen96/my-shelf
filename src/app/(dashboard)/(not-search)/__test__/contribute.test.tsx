import { render } from "@testing-library/react";
import ContributePage from "../contribute/(main)/page";

jest.mock("next/navigation", () => ({
  useRouter: jest.fn(),
}));

describe("Contribute Page", () => {
  it("Should render correctly snapshot", () => {
    expect(render(<ContributePage />)).toMatchSnapshot();
  });
});
