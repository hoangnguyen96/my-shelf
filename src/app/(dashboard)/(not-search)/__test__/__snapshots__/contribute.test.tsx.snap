// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contribute Page Should render correctly snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <p
        class="chakra-text css-1jr2z3f"
      >
        Fill up Book Details
      </p>
      <form
        class="css-79elbk"
      >
        <div
          class="css-92ksox"
        >
          <div
            class="chakra-form-control css-2d82ls"
            role="group"
          >
            <div
              class="chakra-input__group css-bx0blc"
              data-group="true"
            >
              <input
                class="chakra-input css-1x36n8t"
                id="title"
                name="title"
                placeholder="Book name"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="chakra-form-control css-106bepj"
            role="group"
          >
            <div
              class="chakra-select__wrapper css-42b2qy"
            >
              <select
                class="chakra-select css-g3rz38"
                id="field-:r1:"
              >
                <option
                  value=""
                >
                  Category
                </option>
                <option
                  class="css-0"
                  value="SelfHelpBook"
                >
                  SelfHelpBook
                </option>
                <option
                  class="css-0"
                  value="TextBook"
                >
                  TextBook
                </option>
              </select>
              <div
                class="chakra-select__icon-wrapper css-162mkon"
              >
                <svg
                  aria-hidden="true"
                  class="chakra-select__icon"
                  focusable="false"
                  role="presentation"
                  style="width: 1em; height: 1em; color: currentColor;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="chakra-form-control css-2d82ls"
          role="group"
        >
          <div
            class="chakra-input__group css-bx0blc"
            data-group="true"
          >
            <input
              class="chakra-input css-1x36n8t"
              id="author"
              name="author"
              placeholder="Author Name"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="chakra-form-control css-uzlds2"
          role="group"
        >
          <label
            class="chakra-form__label css-zfpjyo"
            data-testid="fileInput"
            for="fileInput"
            id="field-:r3:-label"
          >
            <div
              class="css-mzrecg"
            >
              <svg
                fill="#dfb300"
                height="34"
                viewBox="0 0 50 34"
                width="50"
                xmlns="http://www.w3.org/2000/svg"
                xmlns:xlink="http://www.w3.org/1999/xlink"
              >
                <rect
                  fill="url(#pattern0_2224_1107)"
                  fill-opacity="0.49"
                  height="34"
                  rx="15"
                  width="50"
                />
                <defs>
                  <pattern
                    height="1"
                    id="pattern0_2224_1107"
                    patternContentUnits="objectBoundingBox"
                    width="1"
                  >
                    <use
                      transform="matrix(0.01 0 0 0.0147059 0 -0.235294)"
                      xlink:href="#image0_2224_1107"
                    />
                  </pattern>
                  <image
                    height="100"
                    id="image0_2224_1107"
                    width="100"
                    xlink:href="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAGQAAABkCAYAAABw4pVUAAAAAXNSR0IArs4c6QAABwxJREFUeJztnGtsVEUUx3/dVgSlKGp4GsRGQAtKFKIoBtGgxohRjKhBY0QSAh+sJuIDjQkJBnx/EGtiTDRq8G3ABzGKDyJWfGAUFQQVEaEtkQKtLVBBu344u8l23b1z7t25j23nl0xo2Hmcmf+9d2bOPXPB4XA4HA6Hw+FwOBwOh8PhKA8q4jYgICcA44BBwLHAwMz/HwJagT+A34FtQFccBvZ0jgfmAm8gg5xWplZgNXAvcHrkVvcwUsBM4D3kyteK4JU2AwuQO8yhpBK4AdiEHREKpf3AY8CQiPpUtowHviY8IQoJswjoE0Xnyol+wCPAYaITIzd9B9SG3ssyYTjR3hXF0gFgdsh9TTyTgCbiFyM3PYwsKHodFyHP8FIGrwvYA2zNpEbsrMieo5eJcgHQgf+B2gksA24ExlB4Y1sF1CBL5ieQDWJQUcp14+yLiUA7+oH5F3gFmEqwq7YCOA94Cf93z+IA7ZUVg4Dt6AdkJeIiscUoZLfvR5TrLbafKKqAT9ANQgtwXYi2XAnsUtrSBpwSoi2hMxyYBSwBVgDrkQm3Fd0AfAMMi8jOdUqbGiizSf404EFKd3e8D1RHaHc/YJXStnkR2hWICuTWb6A0EbJpDXBkpD0Q+iAeYZN9e4BjYrBPxTTE3WBDiDTwE3BcpD3ozgDgxwJ25afErboGAcuxJ0Qa6CQZfqRazBvVNkS8RDAF2QnbFCMNLIyyEwYWYLa3LjbrcqgjHE/sBmQ5nBSqgI1427wxNusy3I19IbLp4gj7oWUmZrvHh9W4yVezFLgnYN27gM+BLUjQQdZdUo0EJrQDTwWsO0xSwC+IP6wYnwJrgX3Ab8APmX9DDai4Df9XfCOyIbTp7oiDhfjv+5/Ay4iHoa9tg6YhTj2tMU1IVEgc+4gwGEVpj+K9yBvQwTaMGYLez9MF1JOgpaBFfqb0OXI/8ADiEQjMCmVjHcDVpTRkkZGZZJMXsbd4+RU4P4gR05UNtADnBGkgBEYiwXPbsCvK7dgTJI1sG+7wY0AlsrowVdxO8sTI2mZTlEuxK0g21aP0HM9SVNZFsh5ThcJKbYkyokDdtpLq9fD3iorqLXTUBsXEsClKBd1fNR9EnKCrgXeBb5HVZZeHHV5pqVfjZysqaCIZqymTGDZFmY68jx/qkWc4MB/4AP/izCxW6ZOKwnMDd8seWjFsP760TAA+9GFfK3BSoYpMwWqNxL/p8ytGXKKAzMcHlPatzC98qqLQkrB7YCCoGHGKMhGJI9PYd0luwXmKAnH6pkoVI05RapGXWibbPsstVG/I3Ex8EXy2xIhTlMvR+QUnZwuYJqE3o7O9G7bFiFMU00WfBp7JZt5qyBjH/BGWGHGJMhhz+Ow+4AiA3YaM8yM0HMIXIy5RlihsmpzCHJjWHqKR+YxEwk2jGKgo2wJ4VZFnagqzoyuqc95RD1DUbW5AjkZ4cUYK8x3Q3449nsQhRhxtrzb8ProKEcQrajCKwOc0cI2P/DXAa4Y81yKBB35sCJstht+HViERIQX9KRlG27OnKNszSUunIs8mEhBDlUeL4ffqFGbVkvIyqidwyPB7nxRyJXlRQzzP9p6IcUWbQoK+TIR5mqk3YbqwW1PIsYJ9hoyzKbMTRAllguH3rSnE8bXKkHEMMMOKSb2XAchJYi+2ZK/6FxQVLsZ9pKUULsM8fuuyf1SiO658XxiWBmAsZlvHxmbd/6lAAs9NNnfb89UpChwCzo2iBwbKTZAZmO39Kr/QUehiehuJfxlcToJUo4sTvrVQ4bmKgulMAyNC7ISJchEkBbyF2da/KOK+SgFfKCpIAzuAM0PrijflIEgF8Di6sfQMmhuHPoTlIHKrRb1HSbogRwOvF7ErP7UgX131ZI6ystwJaYrFDplIsiBTkHcf2rGbo634aR+VZtNa4CbCf4eSNEFSSNTIOwq7ctMq8iJ6vMJ7KpFPGV0VwMC/kbmoAfEm70AmLi3NSDRlMcYiX17wYhze7vdheMfsmuiPfCVoEnAF/o+v7QDOwuyS70Zf4G383ymlpkUGu2zcIYti6Fc27aVI8KFpMu5EzoM8a8jn0NOG3FEF73DN6ugfZOKpQx5FjuA0I9+bbCiWwc9ydRlycDFpr0XLhTWI+32DVya/+4f1yGbwLuQUrsNMB3Ancva/2ZQ5yIbuMHIo/mTkDHZbgDp6A4eRs4S1wKPIeycjpeywW4D7gROBm4GPtY32cHYCDyHROrcgy1s1Nj6N1AE8n0kDkbdiFyLLujFEE9cVF11IGNVmZKL+CPiSEqI9ozj30RdxQWe/AqRpM+kbwzYkwLANXYyYmig+HtaZSbsjaMsPTXiLHgsukiRhOEEShhMkYThBEoYTJGE4QRJGkr6Z64cWZDdsyuNwOBwOh8PhcDgcDofD4XA4QuI/DbA67SuZMigAAAAASUVORK5CYII="
                  />
                </defs>
              </svg>
            </div>
          </label>
          <div
            class="chakra-input__group css-bx0blc"
            data-group="true"
          >
            <input
              accept="image/*"
              class="chakra-input css-1vuqmsh"
              id="fileInput"
              type="file"
            />
          </div>
        </div>
        <div
          class="chakra-form-control css-8qt7d5"
          role="group"
        >
          <textarea
            class="chakra-textarea css-1yn82jt"
            id="field-:r4:"
            name="description"
            placeholder="Reason For Your Contribution"
          />
        </div>
        <button
          class="chakra-button css-1qhatj0"
          disabled=""
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <p
      class="chakra-text css-1jr2z3f"
    >
      Fill up Book Details
    </p>
    <form
      class="css-79elbk"
    >
      <div
        class="css-92ksox"
      >
        <div
          class="chakra-form-control css-2d82ls"
          role="group"
        >
          <div
            class="chakra-input__group css-bx0blc"
            data-group="true"
          >
            <input
              class="chakra-input css-1x36n8t"
              id="title"
              name="title"
              placeholder="Book name"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="chakra-form-control css-106bepj"
          role="group"
        >
          <div
            class="chakra-select__wrapper css-42b2qy"
          >
            <select
              class="chakra-select css-g3rz38"
              id="field-:r1:"
            >
              <option
                value=""
              >
                Category
              </option>
              <option
                class="css-0"
                value="SelfHelpBook"
              >
                SelfHelpBook
              </option>
              <option
                class="css-0"
                value="TextBook"
              >
                TextBook
              </option>
            </select>
            <div
              class="chakra-select__icon-wrapper css-162mkon"
            >
              <svg
                aria-hidden="true"
                class="chakra-select__icon"
                focusable="false"
                role="presentation"
                style="width: 1em; height: 1em; color: currentColor;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        class="chakra-form-control css-2d82ls"
        role="group"
      >
        <div
          class="chakra-input__group css-bx0blc"
          data-group="true"
        >
          <input
            class="chakra-input css-1x36n8t"
            id="author"
            name="author"
            placeholder="Author Name"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="chakra-form-control css-uzlds2"
        role="group"
      >
        <label
          class="chakra-form__label css-zfpjyo"
          data-testid="fileInput"
          for="fileInput"
          id="field-:r3:-label"
        >
          <div
            class="css-mzrecg"
          >
            <svg
              fill="#dfb300"
              height="34"
              viewBox="0 0 50 34"
              width="50"
              xmlns="http://www.w3.org/2000/svg"
              xmlns:xlink="http://www.w3.org/1999/xlink"
            >
              <rect
                fill="url(#pattern0_2224_1107)"
                fill-opacity="0.49"
                height="34"
                rx="15"
                width="50"
              />
              <defs>
                <pattern
                  height="1"
                  id="pattern0_2224_1107"
                  patternContentUnits="objectBoundingBox"
                  width="1"
                >
                  <use
                    transform="matrix(0.01 0 0 0.0147059 0 -0.235294)"
                    xlink:href="#image0_2224_1107"
                  />
                </pattern>
                <image
                  height="100"
                  id="image0_2224_1107"
                  width="100"
                  xlink:href="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAGQAAABkCAYAAABw4pVUAAAAAXNSR0IArs4c6QAABwxJREFUeJztnGtsVEUUx3/dVgSlKGp4GsRGQAtKFKIoBtGgxohRjKhBY0QSAh+sJuIDjQkJBnx/EGtiTDRq8G3ABzGKDyJWfGAUFQQVEaEtkQKtLVBBu344u8l23b1z7t25j23nl0xo2Hmcmf+9d2bOPXPB4XA4HA6Hw+FwOBwOh8PhKA8q4jYgICcA44BBwLHAwMz/HwJagT+A34FtQFccBvZ0jgfmAm8gg5xWplZgNXAvcHrkVvcwUsBM4D3kyteK4JU2AwuQO8yhpBK4AdiEHREKpf3AY8CQiPpUtowHviY8IQoJswjoE0Xnyol+wCPAYaITIzd9B9SG3ssyYTjR3hXF0gFgdsh9TTyTgCbiFyM3PYwsKHodFyHP8FIGrwvYA2zNpEbsrMieo5eJcgHQgf+B2gksA24ExlB4Y1sF1CBL5ieQDWJQUcp14+yLiUA7+oH5F3gFmEqwq7YCOA94Cf93z+IA7ZUVg4Dt6AdkJeIiscUoZLfvR5TrLbafKKqAT9ANQgtwXYi2XAnsUtrSBpwSoi2hMxyYBSwBVgDrkQm3Fd0AfAMMi8jOdUqbGiizSf404EFKd3e8D1RHaHc/YJXStnkR2hWICuTWb6A0EbJpDXBkpD0Q+iAeYZN9e4BjYrBPxTTE3WBDiDTwE3BcpD3ozgDgxwJ25afErboGAcuxJ0Qa6CQZfqRazBvVNkS8RDAF2QnbFCMNLIyyEwYWYLa3LjbrcqgjHE/sBmQ5nBSqgI1427wxNusy3I19IbLp4gj7oWUmZrvHh9W4yVezFLgnYN27gM+BLUjQQdZdUo0EJrQDTwWsO0xSwC+IP6wYnwJrgX3Ab8APmX9DDai4Df9XfCOyIbTp7oiDhfjv+5/Ay4iHoa9tg6YhTj2tMU1IVEgc+4gwGEVpj+K9yBvQwTaMGYLez9MF1JOgpaBFfqb0OXI/8ADiEQjMCmVjHcDVpTRkkZGZZJMXsbd4+RU4P4gR05UNtADnBGkgBEYiwXPbsCvK7dgTJI1sG+7wY0AlsrowVdxO8sTI2mZTlEuxK0g21aP0HM9SVNZFsh5ThcJKbYkyokDdtpLq9fD3iorqLXTUBsXEsClKBd1fNR9EnKCrgXeBb5HVZZeHHV5pqVfjZysqaCIZqymTGDZFmY68jx/qkWc4MB/4AP/izCxW6ZOKwnMDd8seWjFsP760TAA+9GFfK3BSoYpMwWqNxL/p8ytGXKKAzMcHlPatzC98qqLQkrB7YCCoGHGKMhGJI9PYd0luwXmKAnH6pkoVI05RapGXWibbPsstVG/I3Ex8EXy2xIhTlMvR+QUnZwuYJqE3o7O9G7bFiFMU00WfBp7JZt5qyBjH/BGWGHGJMhhz+Ow+4AiA3YaM8yM0HMIXIy5RlihsmpzCHJjWHqKR+YxEwk2jGKgo2wJ4VZFnagqzoyuqc95RD1DUbW5AjkZ4cUYK8x3Q3449nsQhRhxtrzb8ProKEcQrajCKwOc0cI2P/DXAa4Y81yKBB35sCJstht+HViERIQX9KRlG27OnKNszSUunIs8mEhBDlUeL4ffqFGbVkvIyqidwyPB7nxRyJXlRQzzP9p6IcUWbQoK+TIR5mqk3YbqwW1PIsYJ9hoyzKbMTRAllguH3rSnE8bXKkHEMMMOKSb2XAchJYi+2ZK/6FxQVLsZ9pKUULsM8fuuyf1SiO658XxiWBmAsZlvHxmbd/6lAAs9NNnfb89UpChwCzo2iBwbKTZAZmO39Kr/QUehiehuJfxlcToJUo4sTvrVQ4bmKgulMAyNC7ISJchEkBbyF2da/KOK+SgFfKCpIAzuAM0PrijflIEgF8Di6sfQMmhuHPoTlIHKrRb1HSbogRwOvF7ErP7UgX131ZI6ystwJaYrFDplIsiBTkHcf2rGbo634aR+VZtNa4CbCf4eSNEFSSNTIOwq7ctMq8iJ6vMJ7KpFPGV0VwMC/kbmoAfEm70AmLi3NSDRlMcYiX17wYhze7vdheMfsmuiPfCVoEnAF/o+v7QDOwuyS70Zf4G383ymlpkUGu2zcIYti6Fc27aVI8KFpMu5EzoM8a8jn0NOG3FEF73DN6ugfZOKpQx5FjuA0I9+bbCiWwc9ydRlycDFpr0XLhTWI+32DVya/+4f1yGbwLuQUrsNMB3Ancva/2ZQ5yIbuMHIo/mTkDHZbgDp6A4eRs4S1wKPIeycjpeywW4D7gROBm4GPtY32cHYCDyHROrcgy1s1Nj6N1AE8n0kDkbdiFyLLujFEE9cVF11IGNVmZKL+CPiSEqI9ozj30RdxQWe/AqRpM+kbwzYkwLANXYyYmig+HtaZSbsjaMsPTXiLHgsukiRhOEEShhMkYThBEoYTJGE4QRJGkr6Z64cWZDdsyuNwOBwOh8PhcDgcDofD4XA4QuI/DbA67SuZMigAAAAASUVORK5CYII="
                />
              </defs>
            </svg>
          </div>
        </label>
        <div
          class="chakra-input__group css-bx0blc"
          data-group="true"
        >
          <input
            accept="image/*"
            class="chakra-input css-1vuqmsh"
            id="fileInput"
            type="file"
          />
        </div>
      </div>
      <div
        class="chakra-form-control css-8qt7d5"
        role="group"
      >
        <textarea
          class="chakra-textarea css-1yn82jt"
          id="field-:r4:"
          name="description"
          placeholder="Reason For Your Contribution"
        />
      </div>
      <button
        class="chakra-button css-1qhatj0"
        disabled=""
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
